### ISSUE TITLE
Convert jupyter notebook to handout

### ISSUE THREAD

**Issue by smith01s** [danijar/handout#33]
A tool or method to convert an .ipynb file into handout equivalent .py would be great.

**Comment by danijar** [comment]
Good idea. This would allow people to prototype in an interactive environment and switch to Handout once they need something more reproducible or want to run locally. I don't see this as part of the Handout Library, maybe better as an input and output format in `nbconvert`. What do you think?

**Comment by epogrebnyak** [comment]
On a sentiment level, Handout is seems to gain ground with people who are using it as lightweight alternative to Jupiter, often as as a one-directional tool. 

As for conversion possibility, I think we should take note we might need a bit more formal specification of Handout format, including an internal representation of the report as a list of blocks and some description of script code parsing rules (would be useful for #34  as well).

**Comment by danijar** [comment]
Good point. The formal specification is the API that's described in the readme:

- Markdown cells translate into triple-quoted comments at top level.
- Calls to `print()` translate into `doc.add_text()` (could be optional).
- `IPython.display.{Image,Video,HTML}` translate into `doc.add_{image,video,html}()`.
- Matplotlib figures translate into `doc.add_figure()`. I'm not sure what logic Jupyter uses to detect figures and whether this logic is available into `nbconvert`?
- The end of a code cell translates into `doc.show()`.

**Comment by gdahia** [comment]
Reviving this thread because I think this would be a very nice feature. Is there anyone working on it?

**Comment by epogrebnyak** [comment]
Not really any work at this point, awaik. Would you like to suggest any
prototype?

With limited functionality this should be easy, perhaps similar to
converting to a script, without graphs, which one can add manually. Do you
think this is a useful conversion?

вт, 27 окт. 2020 г., 12:36 Gabriel Dahia <notifications@github.com>:
